docs/MPURegionsUsageGuide.md

```diff
@@ -132,18 +152,39 @@
 			void up_mpu_set_register(uint32_t *mpu_regs);
 			```
 3. **Example : Adding new MPU region, MPU_REG_NUM_MY_REGION, with RO setting**
-	1. Add your MPU region in enum MPU_REG_NUM in common mpu.h file.
+	1. Add the MPU region type in enum mpu_region_usages_e in common mpu.h file.
 	```c
-	enum MPU_REG_NUM {
+	enum mpu_region_usages_e {
 	 ....
-	 MPU_REG_NUM_MY_REGION,
-	 MPU_REG_NUM_MAX
+	 MPU_REGION_STACKOVF,
+	 MPU_REGION_MYREGION,
+	 MPU_REGION_MAX
 	};
 	```
-	2. To configure this MPU region as RO:
+	2. Add a variable to store the MPU region value in struct mpu_usages_s in common mpu.h file.
+	```c
+        struct mpu_usages_s {
+	 ....
+	 uint8_t nregion_stackovf;
+	 uint8_t nregion_myregion;
+	 uint8_t max_nregion;
+        };
+	```
+
+	3. Initialized the MPU region to its respective number during MPU initialization in mpuinit.c file.
+	```c
+	void mpu_region_initialize(struct mpu_usages_s *mpu) {
+	...
+	 offset += 1;
+	 mpu->nregion_myregion = offset;
+	}
+	```
+	4. To get MPU region number use mpu_get_nregion_info(usages) API.
+	5. To configure this MPU region as RO:
 		1. Use mpu_get_register_value function to obtain mpu register values.
 			- Declare a *regs* array variable (of size 3\*(uint32_t)) and pass it as first argument.
-			- Pass readonly argument as *True* and region as *MPU_REG_NUM_MY_REGION*.
+			- Pass region number obtained in the third step as second argument.
+			- Pass readonly argument as *True*.
 			- Pass proper values for MPU region base address, size and execute permissions.
 			```c
 			mpu_get_register_config_value(regs, MPU_REG_NUM_MY_REGION,  base_address, size, True,  execute);
```

Comment:
We must not use MPU_REG_NUM_MY_REGION here. 
Can you pls show the actual code  for obtaining the region number and holding it in a variable and passing that variable to the get_register api?